Solutions/CiscoSEG/Parsers/CiscoSEGEvent.txt

```diff
@@ -0,0 +1,309 @@
+// Usage Instructions: 
+// Paste below query in log analytics, click on Save button and select as Function from drop down by specifying function name as CiscoSEGEvent and specifying Legacy Category.
+// This parser covers the following types of Cisco SEG logs: Textmail, Delivery, Bounce, HTTP, AMP, Authentication, Consolidated logs.
+// Function usually takes 10-15 minutes to activate. You can then use function alias from any other queries (e.g. CiscoSEGEvent | take 10).
+// Reference : Using functions in Azure monitor log queries : https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+let EventData = CiscoSEG_CL 
+| extend EventVendor = 'Cisco'
+| extend EventProduct = 'SEG';
+let cisco_seg_txt_injection =() {
+EventData
+| extend EventTime = todatetime(extract(@'\A\w{3}\s(\w{3}\s\d{1,2}\s\d{2}:\d{2}:\d{2}\s\d{4})', 1, Message))
+| extend InjectionConnectionId = extract(@'ICID\s=?(\d+)\s', 1, Message)

```

Comment:
Unable to deploy parser. Getting error:  "Failed to resolve table or column or scalar expression named 'Message' ".

Please fix the issue.

Reply:
connector updated